### PR TITLE
Add support for SMOS40 and english names for S1255

### PR DESCRIPTION
**Pull request by elupus**
* Add register for SMOS40 from #32 
* Ignores unnamed coils
* Add english registers for S1255

### CODE REVIEW

File: nibe/console_scripts/convert_csv.py [hunk 1]
```diff
@@ -158,17 +170,25 @@ def calculate_number(register_type: str, register: str):
             return None
 
         if "id" not in self.data:
-            self.data["id"] = self.data["registertype"].combine(
-                self.data["register"], calculate_number
+            id_prefixed = (self.data["title"].str.startswith("id:")) | (
+                self.data["title"] == "-"
```
Comment (elupus) [inline nibe/console_scripts/convert_csv.py:174]: This is rather ugly. There are duplicated registers in some files. These seem to be unnamed and just have a "-" as title or a title like "id:".

I want back and forth, but landed on just ignoring these unnamed registers. If we want them they should be adjusted.

**Comment by elupus** [comment]
Annoying. Seems the exports are dependent on what features are activated. I got one set of parameters from one person, then another export added a few SPA registers. So getting the SMOS40 register set will be an iterative approach, merging CSV files from multiple people.

**Comment by cybermaus** [comment]
If it is helpful, I am still to set the DIP switch for cooling on the S2125, that would probably activate some stuff. 
Originally I was planning on only doing so this summer.

**Comment by elupus** [comment]
@cybermaus yes could be nice to see if we get more register then. (More future proof for you too)

**Comment by dkmh** [comment]
@elupus dont know if it is still useful but here is the register set from my S1255:
[modbus_addresses_all_20221009-06546819317004-2.csv](https://github.com/yozik04/nibe/files/9742382/modbus_addresses_all_20221009-06546819317004-2.csv)

**Comment by elupus** [comment]
I still have some issues with reading and writing coils. padding of data is a bit messy. draft for now.

**Comment by yozik04** [comment]
@elupus only modbus related?

**Comment by elupus** [comment]
Yup

File: nibe/connection/modbus.py [hunk 2]
```diff
@@ -24,6 +25,17 @@ def split_modbus_data(coil: Coil):
...
+    """Split data into chunks of a certain max length, cropping of trailing data."""
+    count = len(data) // chunks
+    res = []
+    for i in range(0, len(data), count):
+        chunk = data[i : i + count]
+        assert all(x == 0 for x in chunk[max_len:])
+        res.append(chunk[0:max_len])
+    return res
+
+
```
Comment (elupus) [inline nibe/connection/modbus.py:38]: This is rather silly. I would like to use Construct here to handle the packing of data. So we can pack a uint32 into two 16bit blocks and and a list of uint32 into a list of uint16.

Would also be nice to handle packing a list of bits, into bits packed into uint8's. To get this to work, we would need to move the raw/non_raw parsing away from coil.py to the connection instead.

Comment (yozik04) [reply nibe/connection/modbus.py:38]: I was thinking that it would hit us later. I think I can do it in upcoming days.

**Comment by elupus** [comment]
I think this should be okey now as an initial step. I've not tested against an actual modbus server, just the assumed call results, but it's closer to correct.

**Comment by elupus** [comment]
@dkmh turns out your dump had a few additional registers. So strange that they are incomplete and different from pump to pump. They must be filtering on some enabled features.

**Comment by cybermaus** [comment]
Could also be firmware version differences. I only recently installed my S40, it immediately requested an update, and in the release text were mentioned fixed modbus registers

File: nibe/connection/modbus.py (hunk 2 above)
Comment (yozik04) [reply nibe/connection/modbus.py:38]: I have moved it to a separate class.

**Comment by yozik04** [comment]
Now I feel it is ready for merge.

**Comment by yozik04** [comment]
@elupus Any comments before I merge?

**Review by elupus — COMMENTED** [review]
Looks good

**Comment by elupus** [comment]
Cam you do a release too, so i can start the HA pull for modbus support?

**Comment by yozik04** [comment]
Released in 1.0.0

**Pull request merged**